spec/rspec/mocks/matchers/have_received_spec.rb

```diff
@@ -107,6 +107,16 @@ module Mocks
           }.to_not raise_error
         end
 
+        it "passes if expectations against yielded keyword arguments pass" do
```

Comment:
May I suggest wrapping this in a `required_kw_args_supported?` check and `eval` like we do e.g. [here](https://github.com/rspec/rspec-mocks/blob/e03dd5e1ed216369223d43b181d40668be779763/spec/rspec/mocks/verifying_doubles/expected_arg_verification_spec.rb#L78) not to break earlier Ruby's parser?

Reply:
Thank you!